pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -159,11 +165,23 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 			authInfoName, _ := config.getAuthInfoName()
 			persister = PersisterForUser(config.configAccess, authInfoName)
 		}
-		userAuthPartialConfig, err := getUserIdentificationPartialConfig(configAuthInfo, config.fallbackReader, persister)
-		if err != nil {
-			return nil, err
+
+		if len(config.auth.username) > 0 && len(config.auth.password) > 0 {
```

Comment:
This is on the right track, but I think we want to put it in `getUserIdentificationPartialConfgi`.